### ISSUE TITLE
[EditorPreview] support outer scope expressions

### ISSUE THREAD

**Issue by jasonLaster** [devtools-html/debugger.html#2536]
#2530 added support for only showing a variable in scope. The issue is that it is too agressive and narrows the scope to the nearest block and not the surrounding block. 

In this case, if you pause at the debugger statement and mouse over the `parent` and `child` declaration you'll see that the parser considers them out of scope.

```js
function(protoProps, staticProps) {
    var parent = this;
    var child;

    if (protoProps && _.has(protoProps, 'constructor')) {
      child = function(){ return parent.apply(this, arguments); };
      debugger
    }
}
```

We can TDD this fix in `utils/tests/parser`

**Comment by ryanjduffy** [comment]
Added a test ... and it passed ... 🤔 ... so I created a sample file and loaded it into the debugger. Looks like it's working correctly to me but let me know if i'm misunderstanding.

![image](https://cloud.githubusercontent.com/assets/788456/24782518/633071e0-1b0c-11e7-82c7-b554da73a666.png)

**Comment by jasonLaster** [comment]
This is now fixed in #2672
